cohttp-curl-async/src/cohttp_curl_async.ml

```diff
@@ -0,0 +1,304 @@
+open Async_kernel
+module Time = Core.Time
+module Fd = Async_unix.Fd
+module Clock = Async_unix.Clock
+
+let ( let+ ) x f = Deferred.map x ~f
+
+module Sink = struct
+  type _ t = String : string t | Discard : unit t
+
+  let string = String
+  let discard = Discard
+end
+
+module Source = struct
+  type t = Empty | String of string
+
+  let empty = Empty
+  let string s = String s
+
+  let to_curl_callback t =
+    match t with
+    | Empty -> fun _ -> ""
+    | String s ->
+        let len = String.length s in
+        let pos = ref 0 in
+        fun max_asked ->
+          if !pos >= len then ""
+          else
+            let chunk_len = min (len - !pos) max_asked in
+            let res = String.sub s !pos chunk_len in
+            pos := !pos + chunk_len;
+            res
+end
+
+module Context = struct
+  type fd_events = {
+    fd : Fd.t;
+    mutable read : unit Ivar.t option;
+    mutable write : unit Ivar.t option;
+  }
+
+  type t = {
+    mt : Curl.Multi.mt;
+    wakeners : (Curl.t, Curl.curlCode Ivar.t) Hashtbl.t;
+    all_events : (Unix.file_descr, fd_events) Hashtbl.t;
+    mutable timeout : (unit, unit) Clock.Event.t option;
+  }
+
+  let create () =
+    (* Most of this is taken from https://github.com/ygrek/ocurl/blob/master/curl_lwt.ml *)
```

Comment:
Is this comment a leftover from the port from lwt? should this also preserve the copyright from https://github.com/ygrek/ocurl/blob/master/COPYING?

Reply:
removed it. the async bindings are written from scratch at this point.